### PR TITLE
Extend the supported types with supported precision list

### PR DESCRIPTION
**Pull request by neon60**
The goal to have all the ROCm library a supported precision list eventually.

### CODE REVIEW

**Comment by neon60** [comment]
@saadrahim The PR has been opened.

**Comment by saadrahim** [comment]
This needs to be merged before the .wordlist is added. https://github.com/RadeonOpenCompute/rocm-docs-core/pull/501 just a commit --amend left.

**Comment by neon60** [comment]
> This needs to be merged before the .wordlist is added. [RadeonOpenCompute/rocm-docs-core#501](https://github.com/RadeonOpenCompute/rocm-docs-core/pull/501) just a commit --amend left.

Ah, OK. I can check that too.

File: .wordlist.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+cuSPARSELt
+csv
+DBUILD
+DCMAKE
+deps
+desc
+GeLU
+Gemm
+GoogleTest
+googletest
```
Comment (saadrahim) [inline .wordlist.txt]: only one spelling for GoogleTest should be correct

File: .wordlist.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
+algo
+api
+asynchronicity
+bfloat
+BFLOAT
+cd
+ClippedReLU
+cmake
+const
+cusparseLt
```
Comment (saadrahim) [inline .wordlist.txt]: only one spelling for cuSPARESELt should be correct

File: .wordlist.txt [hunk 3]
```diff
@@ -0,0 +1,40 @@
+algo
```
Comment (saadrahim) [inline .wordlist.txt]: algorithm?

**Review by saadrahim — CHANGES_REQUESTED** [review]
It may be best to disable spell checking for this repo. A lot the words you are finding are likely in code blocks and should not be added to worldlist.

Or perhaps we should revert the enablement of spellchecks on rst files.

File: .wordlist.txt (hunk 3 above)
Comment (vin-huang) [reply .wordlist.txt]: yes, the abbreviation of algorithm

Comment (saadrahim) [reply .wordlist.txt]: > yes, the abbreviation of algorithm

That is not a valid abbreviation. Algorithm should be used in all places. algo is informal.

**Comment by neon60** [comment]
> It may be best to disable spell checking for this repo. A lot the words you are finding are likely in code blocks and should not be added to worldlist.
> 
> Or perhaps we should revert the enablement of spellchecks on rst files.

I do not want to let this go yet. Found a bug in a previous commit. Please merge in my fix:
https://github.com/RadeonOpenCompute/rocm-docs-core/pull/503

After that, I can, test the empty wordlist.txt.

**Comment by vin-huang** [comment]
> After that, I can, test the empty wordlist.txt.

May I know if this happened yet and what the next step will be?

**Comment by neon60** [comment]
> > After that, I can, test the empty wordlist.txt.
> 
> May I know if this happened yet and what the next step will be?

Already did some tests and I found some other problems. If I can't fix those today, we should disable the spellchecking temporary.

@saadrahim we should discuss this during the meeting.

**Comment by neon60** [comment]
@saadrahim I do not think I should fix the following errors, probably we should not force these everywhere:

```
Error: CHANGELOG.md:11 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "### Added"] https://github.com/DavidAnson/markdownlint/blob/v0.28.2/doc/md024.md
Error: docs/reference/api-library.md:1 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "The hipSPARSELt library is org..."] https://github.com/DavidAnson/markdownlint/blob/v0.28.2/doc/md041.md
Error: Failed with exit code: 1
```

File: README.md [hunk 4]
```diff
@@ -24,7 +32,12 @@ Download pre-built packages either from the [ROCm package servers](https://rocm.
 
 ### Bash helper build script
 
-The root of this repository has a helper bash script `install.sh` to build and install hipSPARSELt on Ubuntu with a single command.  It does not take a lot of options and hard-codes configuration that can be specified through invoking cmake directly, but it's a great way to get started quickly and can serve as an example of how to build/install. A few commands in the script need sudo access, so it may prompt you for a password.
+The root of this repository has a helper bash script `install.sh` to build and
+install hipSPARSELt on Ubuntu with a single command.  It does not take a lot of
+options and hard-codes configuration that can be specified through invoking
+cmake directly, but it's a great way to get started quickly and can serve as an
```
Comment (Naraenda) [inline README.md]: Should CMake be spelled with capitalization?

```suggestion
CMake directly, but it's a great way to get started quickly and can serve as an
```

**Review by Naraenda — CHANGES_REQUESTED** [review]

**Review by saadrahim — APPROVED** [review]

**Comment by saadrahim** [comment]
@vin-huang can you please review this documentation change?

**Review by Naraenda — APPROVED** [review]

**Comment by neon60** [comment]
The spellcheck will pass after the following PR is merged in:

https://github.com/RadeonOpenCompute/rocm-docs-core/pull/540

**Pull request merged**